File: weijia_li/week01/js-homework-MTA/js/mta1.js
// var onLine = prompt('Please enter your start line.');
// var onStop = prompt('Please enter your start stop.');
// var offLine = prompt('Please enter your end line.');
// var offStop = prompt('Please enter your end stop.');

var mta = {
  lineStops: {
    N: ['Times Square', '34th', '28th', '23rd', 'Union Square', '8th'],
    L: ['8th', '6th', 'Union Square', '3rd', '1st'],
    6: ['Grand Central', '33rd', '28th', '23rd', 'Union Square', 'Astor Place']
  },

  troughStops: function (line, onStop, offStop) {
    var arr = this.lineStops[line];

    var onIndex = arr.indexOf(onStop);
    var offIndex = arr.indexOf(offStop);

    if (onIndex < offIndex) {
      return arr.slice((onIndex + 1), (offIndex + 1));
    } else {
      return arr.slice(offIndex, onIndex).reverse();
    }

  },

  planTrip: function (onLine, onStop, offLine, offStop) {
    // on the same line
    if (onLine === offLine) {

      var stops1 = this.troughStops(onLine, onStop, offStop);

      var numStops = stops1.length;

      console.log("You must travel through the following stops on the " + onLine + " line: " + stops1.join(', '));
      console.log(numStops + " stops in total.")

    } else { // on the different lines

      var stops1 = this.troughStops(onLine, onStop, 'Union Square');
      var stops2 = this.troughStops(offLine, 'Union Square', offStop);

      var numStops = stops1.length + stops2.length

      console.log("You must travel through the following stops on the " + onLine + " line: " + stops1.join(', '));
      console.log("Change at Union Square.");
      console.log("Your journey continues through the following stops on the " + offLine + " line: " + stops2.join(', '));
      console.log(numStops + " stops in total.")

    }
  },
};

// mta.planTrip(onLine, onStop, offLine, offStop);
// mta.planTrip('6', 'Astor Place', 'L', '6th');
mta.planTrip('N', '34th', '6', '33rd');
// mta.planTrip('L', 'Times Square', 'L', '6th');

File: max_bausher/week04/MTA_Ruby/MTA1.rb
require "pry"

# define data structure of train lines

puts "What train are you starting from? "
$start_line = gets.chomp.to_sym
p $start_line
puts "What station are you starting from? "
$start_station = gets.chomp
p $start_station
puts "What train are you ending at? "
$end_line = gets.chomp.to_sym
p $end_line
puts "What station are you ending at? "
$end_station = gets.chomp
p $end_station

$lines = {
  :N => ["Times Square", "34th Street", "28th Street", "23rd Street", "Union Square", "8th Street"],
  :L => ["8th Street", "6th Street", "Union Square", "3rd Street", "1st Street"],
  :Six => ["Grand Central", "33rd Street", "28th Street", "23rd Street", "Union Square", "Astor Place"]
}

# p $lines[:N].index "Times Square"
#
# p $lines[:N][1..3].reverse

#  function which takes single line and two stops and sorts out:
#  1) how many stops you need to go through
#  2) lists the stops out

def line_trip line, start, stop
    start_index = $lines[line].index start
    end_index = $lines[line].index stop
    if end_index < start_index
        stops = $lines[line][start_index..end_index].reverse
    else
        stops = $lines[line][start_index..end_index]
    end
    stops
end

#  function which takes user input and outputs appropriate response based on
#   feedback from line trip function

def trip_itinerary startline, startstation, endline, endstation

    if startline != endline
        first_leg = line_trip startline, startstation, "Union Square"
        second_leg = line_trip endline, "Union Square", endstation
        puts "You must travel through the following stops  on #{ startline }: #{ first_leg }"
        puts "Change at Union Square"
        puts "You must continue through the following stops  on #{ endline }: #{ second_leg }"
        puts "Your total stops will be: #{ first_leg } #{ second_leg }"

    else
        first_leg = line_trip startline, startstation, endstation
        puts "Your total stops will be #{ first_leg }"
    end

end


trip_itinerary $start_line, $start_station, $end_line, $end_station

 # binding.pry
